runtime/zsr/variant.hpp

```diff
@@ -19,22 +20,23 @@ namespace impl {
  * List of types and their internal representation.
  */
 /* clang-format off */
-#define TYPELIST(_)                             \
-    /* external       internal */               \
-    _(bool,           uint64_t)                 \
-    _(uint8_t,        uint64_t)                 \
-    _(uint16_t,       uint64_t)                 \
-    _(uint32_t,       uint64_t)                 \
-    _(uint64_t,       uint64_t)                 \
-    _(int8_t,         int64_t)                  \
-    _(int16_t,        int64_t)                  \
-    _(int32_t,        int64_t)                  \
-    _(int64_t,        int64_t)                  \
-    _(float,          double)                   \
-    _(double,         double)                   \
-    _(const char*,    std::string)              \
-    _(std::string,    std::string)              \
-    _(Introspectable, Introspectable)
+#define TYPELIST(_)                         \
+    /* external          internal */        \
+    _(bool,              uint64_t)          \
+    _(uint8_t,           uint64_t)          \
+    _(uint16_t,          uint64_t)          \
+    _(uint32_t,          uint64_t)          \
+    _(uint64_t,          uint64_t)          \
+    _(int8_t,            int64_t)           \
+    _(int16_t,           int64_t)           \
+    _(int32_t,           int64_t)           \
+    _(int64_t,           int64_t)           \
+    _(float,             double)            \
+    _(double,            double)            \
+    _(const char*,       std::string)       \
+    _(std::string,       std::string)       \
+    _(zserio::BitBuffer, zserio::BitBuffer) \
```

Comment:
To simplify working with externals, we could think about adding `zsr::Introspectable Variant::readBitBuffer(zsr::Compound)` as a short-cut for 
```cpp
zserio::BitStreamReader r(myVariant.get<zserio::BitBuffer>()); 
zsr::Introspectable myIntrospectable = myCompound.alloc();
myIntrospectable.read(r);
```